### PR TITLE
Do not use the Block.HintSystem boolean for ESP filtering

### PR DESCRIPTION
**Pull request by hughsie**
Fixes https://github.com/fwupd/fwupd/issues/2676

### CODE REVIEW

**Comment by superm1** [comment]
I don't think we should flat out remove it, but instead use it for preferential selection. As in if there is none it's okay, but if there are 3 pick the one with the flag.

**Comment by hughsie** [comment]
> I don't think we should flat out remove it, but instead use it for preferential selection

Good idea; something like that^^?

File: libfwupdplugin/fu-common.c [hunk 1]
```diff
@@ -2457,13 +2458,26 @@ fu_common_get_esp_default (GError **error)
...
+		FuVolume *vol = g_ptr_array_index (volumes, i);
+		g_autofree gchar *type = fu_volume_get_id_type (vol);
+		if (g_strcmp0 (type, "vfat") == 0 &&
+		    fu_volume_is_internal (vol)) {
+			has_internal = TRUE;
+			break;
+		}
+	}
+
 	/* only add in internal vfat partitions */
```
Comment (superm1) [inline libfwupdplugin/fu-common.c:2474]: I think this comment probably needs to be changed

**Review by superm1 — APPROVED** [review]
code changes look good, just update a comment

**Pull request merged**